src/client.rs

```diff
@@ -126,7 +146,13 @@ impl Client {
       .body(Body::from(
         R::body(input)?.unwrap_or_else(|| Cow::Borrowed(&[0; 0])),
       ))?;
-
+    #[cfg(feature = "gzip")]
+    {
+      trace!("Add {}: {}", hyper::header::ACCEPT_ENCODING, GZIP_ENCODING);
+      let header_name = hyper::header::ACCEPT_ENCODING;
+      let header_value = HeaderValue::from_static(GZIP_ENCODING);
+      let _unused = request.headers_mut().insert(header_name, header_value); 
+    }
```

Comment:
I'd rather we structure things a bit differently instead of having all this complexity in-line here. How about the following:
```patch
@@ -105,6 +108,21 @@ impl Client {
     Builder::default().build(api_info)
   }

+  /// Add "gzip" as an accepted encoding to the request.
+  #[cfg(feature = "gzip")]
+  fn maybe_add_gzip_header(request: &mut Request<Body>) {
+    use http::header::ACCEPT_ENCODING;
+    use http::HeaderValue;
+
+    let _ = request
+      .headers_mut()
+      .insert(ACCEPT_ENCODING, HeaderValue::from_static("gzip"));
+  }
+
+  /// Do nothing.
+  #[cfg(not(feature = "gzip"))]
+  fn maybe_add_gzip_header(_request: &mut Request<Body>) {}
+
   /// Create a `Request` to the endpoint.
   fn request<R>(&self, input: &R::Input) -> Result<Request<Body>, R::Error>
   where
@@ -117,7 +135,7 @@ impl Client {
     url.set_path(&R::path(input));
     url.set_query(R::query(input)?.as_ref().map(AsRef::as_ref));

-    let request = HttpRequestBuilder::new()
+    let mut request = HttpRequestBuilder::new()
       .method(R::method())
       .uri(url.as_str())
       // Add required authentication information.
@@ -127,9 +145,55 @@ impl Client {
         R::body(input)?.unwrap_or(Cow::Borrowed(&[0; 0])),
       ))?;

+    Self::maybe_add_gzip_header(&mut request);
     Ok(request)
   }
```

What do you think?

Reply:
I think this is great. I honestly had not thought to define the method in both cases, which is why I was using the cfg around the block. This is much cleaner, thank you.